openapi/openapi.json

```diff
@@ -18263,6 +18263,52 @@
         }
       }
     },
+    "com.github.openshift.api.machine.v1alpha1.AdditionalBlockDevice": {
+      "description": "AdditionalBlockDevice is a block device to attach to the server.",
+      "type": "object",
+      "required": [
+        "type",
+        "name",
+        "diskSize"
+      ],
+      "properties": {
+        "availabilityZone": {
+          "description": "AvailabilityZone is the volume availability zone to create the volume in. This can only be specified when the type is \"volume\". If omitted, the availability zone of the server will be used.",
+          "type": "string"
+        },
+        "diskSize": {
+          "description": "Size is the size in GB of the block device.",
+          "type": "integer",
+          "format": "int32",
+          "default": 0
+        },
+        "name": {
+          "description": "Name of the block device in the context of a machine. If the block device is a volume, the Cinder volume will be named as a combination of the machine name and this name. Also, this name will be used for tagging the block device.",
+          "type": "string",
+          "default": ""
+        },
+        "type": {
+          "description": "Type is the type of block device to create. This can be either \"volume\" or \"local\".",
+          "type": "string",
+          "default": ""
+        },
+        "volumeType": {
+          "description": "VolumeType is the volume type of the volume. This can only be specified when the type is \"volume\". If omitted, the default type will be used.",
+          "type": "string"
+        }
+      },
+      "x-kubernetes-unions": [
+        {
+          "discriminator": "type",
+          "fields-to-discriminateBy": {
+            "availabilityZone": "AvailabilityZone",
+            "diskSize": "Size",
+            "name": "Name",
+            "volumeType": "VolumeType"
+          }
+        }
+      ]
```

Comment:
Oh, interesting. What generated that?

From https://github.com/kubernetes/enhancements/issues/1027 it reads like the effort had stalled. Will the APIserver respect them?

Reply:
It won't for a raw extension object like MAPI providerSpec, but the idea is that in the future yes it will. We implement the validation with CEL until then